src/codegen/cfg.rs

```diff
@@ -337,6 +338,7 @@ pub struct ControlFlowGraph {
     pub ty: pt::FunctionTy,
     pub selector: u32,
     current: usize,
+    pub array_lengths_temps: ArrayTempVars, //a new member in the cfg that track the temp vars of arrays
```

Comment:
Please call it `array_length_vars`. The fact that they're temporaries is an implementation detail.